proposals/gc/MVP.md

```diff
@@ -186,34 +203,32 @@ Perhaps add the following short-hands:
 * `struct.new <typeidx>` allocates a structure of type `$t` and initialises its fields with given values
   - `struct.new $t : [t*] -> [(ref $t)]`
     - iff `$t = struct (mut t)*`
-  - equivalent to `struct.new_with_rtt $t (rtt.canon any)`
 
 * `struct.new_with_rtt <typeidx>` allocates a structure of type `$t` with RTT information determining its [runtime type](#values) and initialises its fields with given values
-  - `struct.new_with_rtt $t : [(rtt t') t*] -> [(ref $t)]`
+  - `struct.new_with_rtt $t : [(rtt n t') t*] -> [(ref $t)]`
     - iff `$t = struct (mut t)*`
-    - and `(type $t) <: t'`
+    - and `(type $t) == t'`
```

Comment:
> I'm confused by the (type $t) part. Isn't $t a type?

It's a "type use" referring to `$t`, see the new definition of [heap types](https://github.com/WebAssembly/function-references/blob/master/proposals/function-references/Overview.md#types). The extra keyword avoids ambiguity in some syntactic contexts and is more consistent with other syntax. Note that `(ref $t)` is defined as a shorthand for `(ref (type $t))`.

FWIW, I just removed the type annotation on the instruction, because with type equivalence it can be derived from the operand type.

> If I get to make further wishes: I find the t* part confusing too, because that's not a repeated sequence of a single t type.

It's a sequence of possibly different value types. It's a notational convention that is used heavily in the spec, explained [here](https://webassembly.github.io/spec/core/syntax/conventions.html#auxiliary-notation).

Reply:
> I just removed the type annotation on the instruction, because with type equivalence it can be derived from the operand type.

I generally agree with removing the redundancy; but I think the current specification `[(rtt n t) t'*]` isn't implementable: we can't find the rtt on the value stack if we don't know how many `t'` there are, which in turn we can only determine from the rtt. I can see several solutions:
(1) go back to having a type immediate
(2) add an arity immediate (equalling `n`, indicating the number of `t'` stack values)
(3) swap the order of the stack arguments to `[t'* (rtt n t)]`

Reply:
Ah, yes, good catch, that's ambiguous. I reinstated the type immediate. I figure that if we want to remove it, we should do so consistently along with the other instructions anyway.